### DISCUSSION TITLE
Make HealthChecks configurable

### DISCUSSION THREAD

**Discussion by makakken** [directus/directus#22588] [Feature Requests]
### Summary

the /server/health API endpoint serves different metrics. it would be nice to turn on/off which ones should be checked for the health-check to be OK.

### Motivation

I've setup the smtp-connectiion for directus. the health-check includes opening a smtp-connection each time the health-check url is called it opens an smtp-connection to check if the connection is available. my mail-host rate-limited me because of regular health-checks every 60 seconds. so i would love to disable smtp-health-checking

### Detailed Design

The Implementation should add a configuration-page in directus backend, or at least some configuartion variables to switch off health-checking for specific services (eg.: smtp, s3, etc)

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

if the selected features are not included in health-checking, you are not aware of they're ability to work as expected.

### Alternatives

Increasing the health-check intervall, so you're not getting into any services rate-limit.

### Adoption Strategy

beacause it's a new env-variable or config-page, you will not affect existing directus users. default is that every service configured is health-checked. so simply health-check them by default, but give users the ability to turn off, if wanted.

**Comment by BillyBouman-2B-IT** [comment]
Were you using Google SMTP relay perhaps? I think I'm experiencing the same.

```json
{
    "status": "error",
    "releaseId": "11.1.1",
    "serviceId": "---REDACTED---",
    "checks": {
        "pg:responseTime": [
            {
                "status": "ok",
                "componentType": "datastore",
                "observedUnit": "ms",
                "observedValue": 2.95,
                "threshold": 150
            }
        ],
        "pg:connectionsAvailable": [
            {
                "status": "ok",
                "componentType": "datastore",
                "observedValue": 2
            }
        ],
        "pg:connectionsUsed": [
            {
                "status": "ok",
                "componentType": "datastore",
                "observedValue": 0
            }
        ],
        "cache:responseTime": [
            {
                "status": "ok",
                "componentType": "cache",
                "observedValue": 3.017,
                "observedUnit": "ms",
                "threshold": 150
            }
        ],
        "storage:local:responseTime": [
            {
                "status": "ok",
                "componentType": "objectstore",
                "observedValue": 3.784,
                "observedUnit": "ms",
                "threshold": 750
            }
        ],
        "email:connection": [
            {
                "status": "error",
                "componentType": "email",
                "output": {
                    "code": "ECONNECTION",
                    "response": "421-4.7.0 Try again later, closing connection. (EHLO)\n421-4.7.0  For more information, go to\n421 4.7.0  https://support.google.com/a/answer/3221692 a640c23a62f3a-aa50b52b7b2sm58508866b.133 - gsmtp",
                    "responseCode": 421,
                    "command": "EHLO"
                }
            }
        ]
    }
}
```
